### PR TITLE
TNT-43002 Fix duplicate inline scripts added to DOM

### PR DESCRIPTION
**Pull request by XDex**
## Description

When HTML offers with inline SCRIPT tags are rendered, these are added twice to page DOM, even though the scripts are actually executed just once on the page. We should fix it, so that no unexpected inline SCRIPT tags are added to page DOM after Personalization offers are applied.

## Related Issue

https://jira.corp.adobe.com/browse/TNT-43002

## Motivation and Context

When HTML offers with inline SCRIPT tags are rendered, these are added twice to page DOM, even though the scripts are actually executed just once on the page. We should fix it, so that no unexpected inline SCRIPT tags are added to page DOM after Personalization offers are applied.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [x] I have made any necessary test changes and all tests pass.
- [x] I have run the Sandbox successfully.

### CODE REVIEW

File: src/components/Personalization/dom-actions/scripts.js [hunk 1]
```diff
@@ -80,8 +80,11 @@ export const getRemoteScriptsUrls = fragment => {
   return result;
 };
 
-export const executeInlineScripts = (container, scripts, func) => {
-  scripts.forEach(script => func(container, script));
+export const executeInlineScripts = (parent, scripts) => {
+  scripts.forEach(script => {
+    parent.appendChild(script);
+    parent.removeChild(script);
```
Comment (jonsnyder) [inline src/components/Personalization/dom-actions/scripts.js:86]: How can you be sure the inline script is done executing? Do scripts that attach event handlers still work?

File: test/unit/specs/components/Personalization/dom-actions/setHtml.spec.js [hunk 2]
```diff
@@ -49,13 +49,16 @@ describe("Personalization::actions::setHtml", () => {
...
       meta
     };
     const event = { elements };
 
     return setHtml(settings, event).then(() => {
       expect(window.someEvar123).toEqual(1);
+
+      const scriptElements = document.querySelectorAll("#evar123");
+      expect(scriptElements.length).toEqual(1);
     });
```
Comment (jonsnyder) [inline test/unit/specs/components/Personalization/dom-actions/setHtml.spec.js:62]: Can you add a test to see if an event handler defined in the inlineScript still works? And what if you have a setTimeout in the inline script?

**Review by jonsnyder — CHANGES_REQUESTED** [review]
Can you add an integration test?

**Comment by XDex** [comment]
> Can you add an integration test?

@jonsnyder I don't think an integration test is necessary, as we're only changing the rendering part here, which can be fully tested via unit tests, which I've added.

Note that I've also added some changes to unit test runner config, so async/await is now supported in unit tests.

**Comment by jfkhoury** [comment]
@XDex Thank you did the changes to the unit tests and for making this change; a cleaner approach even if we weren’t double executing those script tags! 

I’m not sure how the unit test is verifying that the script is being removed. I might have missed it. 

One last thing, If @jonsnyder believes an integration test is helpful, let’s add one please; being extra safe never hurts, especially with those complicated Personalization features. No rush on this one since we can’t release anything anyway. 

Thanks again, it’s a pleasure to have you contribute to the project.

**Comment by XDex** [comment]
@jfkhoury Thanks! I'll add a functional test as you suggest..
> I’m not sure how the unit test is verifying that the script is being removed. I might have missed it.

This is checked at line 62 in the test: `    expect(scriptElements.length).toEqual(1);`

**Comment by ninaceban** [comment]
@XDex we have discussed about Alloy vs at.js in the past. Just because at.js does something in particular way, it doesn't mean that Alloy must follow suit.
I would argue that having scripts being part of the DOM both during `innerHTML` operation and after `appendChild` allows us to clearly see which DOM operation contained scripts and must be "treated" a little bit differently and it helps with debugging.

I would let @jfkhoury and @jonsnyder share their thoughts, but I think just documenting the behavior should be enough. We don't need to add more complexity to the existing DOM manipulation code.

**Comment by XDex** [comment]
@ninaceban I have provided the motivation for this change in the linked [Jira ticket](https://jira.corp.adobe.com/browse/TNT-43002) comments, and I think @jfkhoury agrees that we should have this fixed.
@jfkhoury @jonsnyder Please take another look, I've also added an integration test.. Thank you!

**Review by jonsnyder — APPROVED** [review]
Thanks for adding the unit tests!

**Review by jfkhoury — APPROVED** [review]
Thank you for the tests.

**Pull request merged**